controllers/apps/transformer_rbac.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright (C) 2022-2023 ApeCloud Co., Ltd
+
+This file is part of KubeBlocks project
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+package apps
+
+import (
+	"fmt"
+
+	"github.com/spf13/viper"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/apecloud/kubeblocks/internal/controller/builder"
+	"github.com/apecloud/kubeblocks/internal/controller/graph"
+	ictrltypes "github.com/apecloud/kubeblocks/internal/controller/types"
+)
+
+// RBACTransformer puts the rbac at the beginning of the DAG
+type RBACTransformer struct{}
+
+var _ graph.Transformer = &RBACTransformer{}
+
+func (c *RBACTransformer) Transform(ctx graph.TransformContext, dag *graph.DAG) error {
+	if !viper.GetBool("ENABLE_RBAC_MANAGER") {
+		return nil
+	}
+
+	transCtx, _ := ctx.(*ClusterTransformContext)
+	cluster := transCtx.Cluster
+	if cluster.IsDeleting() {
+		return nil
+	}
```

Comment:
I've recorded that deleting path has been optimized, don't need to have this check block, @free6om plz review and perhaps check if there are similar pattern elsewhere.